File: API/Entities/RaffleEntry.cs
namespace API.Entities;

public class RaffleEntry : DeletableEntity
{
    public required string Id { get; set; }

    public required string RaffleEventId { get; set; }
    public RaffleEvent? RaffleEvent { get; set; }

    public required string ParticipantId { get; set; }
    public EventParticipant? Participant { get; set; }

    public int Donation { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public int LowTicket { get; set; }
    public int HighTicket { get; set; }

    public bool Complimentary { get; set; }
}

File: client/src/data/entities/raffle-entry.ts
import { EventParticipant } from './event-participant.ts';

export interface RaffleEntry {
  id: string;
  participant: EventParticipant;
  donation: number;
  lowTicket: number;
  highTicket: number;
  complimentary: boolean;
}

export interface NewRaffleEntry {
  eventId: string;
  participantId: string;
  donation: number;
  complimentary: boolean;
}